### DISCUSSION TITLE
How to display different templates for different property value?

### DISCUSSION THREAD

**Discussion by kochkaqqq** [AvaloniaUI/Avalonia#14302] [Q&A]
In depends on model state, I need to use different control templates. Now my MainWindow.xaml looks like this:
`
< Grid ColumnDefinitions="*, *, *" RowDefinitions="*, *" >
		< StackPanel Grid.ColumnSpan="3" Grid.Row="0" Grid.Column="0" Background="LightGoldenrodYellow" >
			<controls:FirstStateTemp VerticalAlignment="Center" IsVisible="{Binding IsFirstState}" />
			<controls:SecondStateTemp VerticalAlignment="Center" IsVisible="{Binding IsSecondState}" />
		</ StackPanel>

	<Button Command="{Binding SerFirstState}" Grid.Row="1" Grid.Column="0" Content="change #1" 
                      HorizontalAlignment="Center" />
	<Button Command="{Binding SetSecondState}" Grid.Row="1" Grid.Column="1" Content="change #2" 
                       HorizontalAlignment="Center" />
	<Button Command="{Binding Hide}" Grid.Row="1" Grid.Column="2" Content="hide" HorizontalAlignment="Center" />
</ Grid>
`

And MainWindowViewModel.cs:

public bool IsFirstState {...}
public bool IsSecondState {...}

private fields
...
commands
...

Are there any ways to do it better? Maybe there is some kind of tag that can use different template by different property value?

**Comment by jp2masa** [comment]
https://docs.avaloniaui.net/docs/basics/data/data-templates
https://docs.avaloniaui.net/docs/concepts/templates

**Reply by thevortexcloud** [reply]
To add onto this, you can create a custom view locator.

https://docs.avaloniaui.net/docs/concepts/view-locator
https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/IDataTemplateSample
